core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -59,7 +60,8 @@ class LogSegment private[log] (val log: FileRecords,
                                val baseOffset: Long,
                                val indexIntervalBytes: Int,
                                val rollJitterMs: Long,
-                               val time: Time) extends Logging {
+                               val time: Time,
+                               val needsFlushParentDir: Boolean = false) extends Logging {
```

Comment:
Could we add the new param to the javadoc?

Reply:
Done.